From 3ab0d908ca6e806e75fe477ae6f0f35abc6ff128 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 25 Nov 2019 19:08:48 +0800
Subject: [PATCH 24/60] waylandsink: Commit the parent after creating
 subsurface

We should commit the parent to activate new subsurface, this is
documented in the protocol.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
(cherry picked from commit 207706133312ea070d909bd571c0a9388aae6292)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 ext/wayland/wlwindow.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ext/wayland/wlwindow.c b/ext/wayland/wlwindow.c
index b226d5034..0c001aa1a 100644
--- a/ext/wayland/wlwindow.c
+++ b/ext/wayland/wlwindow.c
@@ -342,6 +342,8 @@ gst_wl_window_new_in_surface (GstWlDisplay * display,
       window->area_surface, parent);
   wl_subsurface_set_desync (window->area_subsurface);
 
+  wl_surface_commit (parent);
+
   return window;
 }
 
-- 
2.20.1

